### ISSUE TITLE
TLS not working in OCP cluster with self signed cert.

### ISSUE THREAD

**Issue by whitingjr** [Hyperfoil/horreum-operator#5]
A deployed Horreum replica is reporting this error in the log.

2023-03-23 19:34:29,728 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-eventloop-thread-1) HTTP Request to /api/user/administrators failed, error id: 086eb16d-d056-4b99-a1e0-6b8ff8122923-1: java.util.concurrent.CompletionException: javax.ws.rs.WebApplicationException: HTTP 500 Internal Server Error
712at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
713at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)

Caused by: javax.net.ssl.SSLHandshakeException: Failed to create SSL connection
1059at io.vertx.core.net.impl.ChannelProvider$1.userEventTriggered(ChannelProvider.java:130)
1060... 25 more

**Comment by rafaeltuelho** [comment]
adding some additional info related to this...

I see the Operator [1] sets the quarkus [2] `QUARKUS_OIDC_TLS_VERIFICATION` to `none` which means it should not verify the TLS certificate used by the OIDC server (Keycloak in this case). Maybe it worth investigate if this issue is really related to the communication between horreum backend and keycloak through the Quarkus OIDC extension. 

[1] https://github.com/Hyperfoil/horreum-operator/blob/104e3a4f23924714b25cd269017edf6cf8f2a713/controllers/app.go#L38
[2] Quarkus OIDC extension reference: https://quarkus.io/guides/security-openid-connect

**Comment by johnaohara** [comment]
The error comes from a request to rest endpoint `/api/user/administrators`. The exception from a backend call is wrapped here: https://github.com/Hyperfoil/Horreum/blob/0.7/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/UserServiceImpl.java#L372
when an exception occurs from this call: https://github.com/Hyperfoil/Horreum/blob/0.7/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/UserServiceImpl.java#L452

the client call to keycloak is throwing the exception: https://github.com/Hyperfoil/Horreum/blob/0.7/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/UserServiceImpl.java#L448

it might be that `QUARKUS_OIDC_TLS_VERIFICATION=none` is no longer working as expected, or there is something else broken

**Comment by dustinblack** [comment]
I attempted as a test to set `QUARKUS_TLS_TRUST_ALL =  true` as and environment variable for the `horreum-app` pod, but I got this:

```
2023-08-16 13:27:18,627 WARN  [io.qua.run.con.ConfigRecorder] (main) Build time property cannot be changed at runtime:
 - quarkus.tls.trust-all is set to 'true' but it is build time fixed to 'false'. Did you change the property quarkus.tls.trust-all aft
er building the application?
```

**Comment by dustinblack** [comment]
@barreiro This may be a useful relevant reference: https://developers.redhat.com/blog/2017/11/22/dynamically-creating-java-keystores-openshift

**Comment by johnaohara** [comment]
It looks like we should be able to register a truststore for the OIDC client: https://github.com/quarkusio/quarkus/pull/18012

I haven't looked at the mechanics of how to do it though

**Comment by barreiro** [comment]
In this case, the problem is not the OIDC client. 

Horreum uses [Keycloak admin client](https://github.com/Hyperfoil/Horreum/blob/0.7.9/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/UserServiceImpl.java#L80) directly. A resteasy client is not provided, so a new one is created with defaults.

We would have to provide a resteasy client with the right trust store there.

In the scope of the operator, an alternative would be to add the `service-ca` certificate to the system-wide trust store, so that all SSL connections to other services are trusted. See #17.

**Comment by barreiro** [comment]
fixed in #17
